db/migrate/20231215163154_create_guests.rb

```diff
@@ -0,0 +1,11 @@
+class CreateGuests < ActiveRecord::Migration[7.0]
+  def change
+    create_table :guests do |t|
+      t.string :name
+      t.integer :nights
+      t.references :room, foreign_key: true
```

Comment:
For a many-to-many relationship, we do *not* want the room_id on the guests table. A guest can stay in many different types of rooms, not just one. So it would be hard to track all of that in a foreign key here. Instead, all our foreign keys would live on our joins table (where we could have room_id and guest_id to associate the 2 to indicate one guest stay)